crates/bevy_asset/src/io/embedded/mod.rs

```diff
@@ -111,13 +111,28 @@ macro_rules! embedded_path {
     }};
 
     ($source_path: expr, $path_str: expr) => {{
-        let crate_name = module_path!().split(':').next().unwrap();
-        let after_src = file!().split($source_path).nth(1).unwrap();
-        let file_path = std::path::Path::new(after_src)
-            .parent()
-            .unwrap()
-            .join($path_str);
-        std::path::Path::new(crate_name).join(file_path)
+        let crate_name = module_path!()
+            .split(':')
+            .next()
+            .expect("Could not find crate name");
```

Comment:
Can we report the original string here for better diagnosability?

Reply:
Certainly. Is this the kind of error message you're looking for?

```
    panic!("Could not find crate name. Expected ':' in module path `{}`", module_path!());
```